packages/utils/modify-config-helper.ts

```diff
@@ -87,59 +96,94 @@ export default function modifyHelperUtil(
 			}
 		};
 	}
-	env.registerStub(generator, generatorName);
 
-	env.run(generatorName)
-		.then(
-			(): void => {
-				let configModule: object;
-				try {
-					const confPath = path.resolve(process.cwd(), ".yo-rc.json");
-					configModule = require(confPath);
-					// Change structure of the config to be transformed
-					const tmpConfig: object = {};
-					Object.keys(configModule).forEach(
-						(prop: string): void => {
-							const configs = Object.keys(configModule[prop].configuration);
-							configs.forEach(
-								(conf: string): void => {
-									tmpConfig[conf] = configModule[prop].configuration[conf];
-								}
-							);
-						}
-					);
-					configModule = tmpConfig;
-				} catch (err) {
-					console.error(chalk.red("\nCould not find a yeoman configuration file.\n"));
-					console.error(
-						chalk.red(
-							"\nPlease make sure to use 'this.config.set('configuration', this.configuration);' at the end of the generator.\n"
-						)
-					);
-					Error.stackTraceLimit = 0;
-					process.exitCode = -1;
+	if (!usingCLI) {
+		/**
```

Comment:
no need to add comment here, you will need to disable prompting and fully generate a scaffold file by running the transformation on the config object

Reply:
Not exactly, I meant to use `prompting` but instead of directly using `this.prompt` which internally uses inquirer questions, I meant to overload `this.prompt` with prompting on the socket.

Rest of the work, like running transformations, generate scaffold, etc. is already implemented for CLI and would be reused.

Is this approach is OK for you?